src/trix/elements/trix_editor_element.js

```diff
@@ -161,6 +161,17 @@ installDefaultCSSForTagName("trix-editor", `\
 }`)
 
 export default class TrixEditorElement extends HTMLElement {
+  static formAssociated = "ElementInternals" in window
+
+  #customValidationMessage
+  #internals
+
+  constructor() {
+    super()
+    this.#internals = this.constructor.formAssociated ?
+      this.attachInternals() :
+      null
```

Comment:
Instead of repeating the if/else structure in the corresponding methods, how do you feel about making that decision here and keep both paths separated. I haven't tested this but something like:

```js
export default class TrixEditorElement extends HTMLElement {
  constructor() {
    // ...
    if (!this.#internals) {
      this.#disableElementInternals()
    }
  }
  
  get disabled() {
    return this.inputElement.disabled
  }

  #disableElementInternals() {
    Object.assign(this, {
      get disabled() {
        console.warn("This browser does not support the [disabled] attribute for trix-editor elements.")
      }

      // ...
    })
  }
}
```

Open to other approaches, but I'd love to keep both paths separated: the default path where it assumes support and the alternative "not supported" path.

Reply:
I'd originally proposed https://github.com/basecamp/trix/pull/1132 to extract a delegate object to serve as a unified grouping of either the `if` branch's code or the `else` branch's code.

Knowing that object delegation in that style has fallen out of favor, I decided an explicit (albeit verbose) repetition of the conditional in each method would be easier to review.

Would something like this work:

```js
class LegacyDelegate {
  constructor(element) {
    this.element = element
  }
  // ...
}

class ElementInternalsDelegate {
  constructor(element) {
    this.element = element
    this.internals = element.attachInternals()
  }
  // ...
}

class TrixEditorElement extends HTMLElement {
  static formAssociated = "ElementInternals" in window

  #delegate 

  constructor() {
    this.#delegate = this.constructor.formAssociated ?
      new ElementInternalsDelegate(this) :
      new LegacyDelegate(this)
  }
}
```

Reply:
Yes @seanpdoyle, that would work for me 👍